sdk/apis/apis/v1alpha2/types_apiexport_conversion.go

```diff
@@ -0,0 +1,186 @@
+// resource schemas present in the APIExport's spec. To have a full, lossless conversion, use
+// Convert_v1alpha2_APIExport_To_v1alpha1_APIExport instead.
+func Convert_v1alpha2_APIExportSpec_To_v1alpha1_APIExportSpec(in *APIExportSpec, out *apisv1alpha1.APIExportSpec, s kubeconversion.Scope) error {
+	if in.Identity != nil {
+		out.Identity = &apisv1alpha1.Identity{}
+		if err := Convert_v1alpha2_Identity_To_v1alpha1_Identity(in.Identity, out.Identity, s); err != nil {
+			return err
+		}
+	}
+
+	if in.MaximalPermissionPolicy != nil {
+		out.MaximalPermissionPolicy = &apisv1alpha1.MaximalPermissionPolicy{}
+		if err := Convert_v1alpha2_MaximalPermissionPolicy_To_v1alpha1_MaximalPermissionPolicy(in.MaximalPermissionPolicy, out.MaximalPermissionPolicy, s); err != nil {
+			return err
+		}
+	}
+
+	if claims := in.PermissionClaims; claims != nil {
+		newClaims := []apisv1alpha1.PermissionClaim{}
+		for _, claim := range claims {
+			var newClaim apisv1alpha1.PermissionClaim
+			if err := Convert_v1alpha2_PermissionClaim_To_v1alpha1_PermissionClaim(&claim, &newClaim, s); err != nil {
+				return err
+			}
+			newClaims = append(newClaims, newClaim)
+		}
+		out.PermissionClaims = newClaims
+	}
+
+	latest, _ := convert_v1alpha2_ResourceSchemas_To_v1alpha1_LatestResourceSchemas(*in)
+	if len(latest) > 0 {
+		out.LatestResourceSchemas = latest
+	}
+
+	return nil
+}
+
+func Convert_v1alpha1_APIExport_To_v1alpha2_APIExport(in *apisv1alpha1.APIExport, out *APIExport, s kubeconversion.Scope) error {
+	if err := autoConvert_v1alpha1_APIExport_To_v1alpha2_APIExport(in, out, s); err != nil {
+		return err
+	}
+
+	if overhanging, ok := in.Annotations[resourceSchemasAnnotation]; ok {
+		resourceSchemas := []ResourceSchema{}
+		if err := json.Unmarshal([]byte(overhanging), &resourceSchemas); err != nil {
+			return fmt.Errorf("failed to decode schemas from JSON: %w", err)
+		}
+
+		if len(resourceSchemas) > 0 {
+			if out.Spec.ResourceSchemas == nil {
+				out.Spec.ResourceSchemas = []ResourceSchema{}
+			}
+
+			out.Spec.ResourceSchemas = append(out.Spec.ResourceSchemas, resourceSchemas...)
+		}
+
+		delete(out.Annotations, resourceSchemasAnnotation)
+
+		// make tests for equality easier to write by turning []string into nil
+		if len(out.Annotations) == 0 {
+			out.Annotations = nil
+		}
+	}
+
+	return nil
+}
+
+func Convert_v1alpha1_APIExportSpec_To_v1alpha2_APIExportSpec(in *apisv1alpha1.APIExportSpec, out *APIExportSpec, s kubeconversion.Scope) error {
+	if in.Identity != nil {
+		if err := Convert_v1alpha1_Identity_To_v1alpha2_Identity(in.Identity, out.Identity, s); err != nil {
+			return err
+		}
+	}
+
+	if in.MaximalPermissionPolicy != nil {
+		if err := Convert_v1alpha1_MaximalPermissionPolicy_To_v1alpha2_MaximalPermissionPolicy(in.MaximalPermissionPolicy, out.MaximalPermissionPolicy, s); err != nil {
+			return err
+		}
+	}
+
+	if claims := in.PermissionClaims; claims != nil {
```

Comment:
this is untested

Reply:
By untested, you mean if it's being converted roundtrip?

Reply:
Yes. If I get these lines wrong, no unit test will fail (before roundtrip tests).